hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -764,6 +764,56 @@ object LowerTableIR {
                 ++ rValueFields.map(f => f -> GetField(rEltRef, f)))
           })
 
+        case x@TableUnion(children) =>
+          val lowered = children.map(lower)
+          val keyType = x.typ.keyType
+          val newPartitioner = RVDPartitioner.generate(keyType, lowered.flatMap(_.partitioner.rangeBounds))
+          val repartitioned = lowered.map(_.repartitionNoShuffle(newPartitioner))
+
+          TableStage(
+            repartitioned.flatMap(_.letBindings),
+            repartitioned.flatMap(_.broadcastVals),
```

Comment:
I think the globals from the tail children will be here, but should get pruned away.

Reply:
Don't they have to be here, because the globals could be referenced in the tail children's bodies?

Reply:
Possible. I always forget the semantics of globals and broadcast vals when I haven't looked at it in the last day or two.